syft/frameworks/torch/tensors/interpreters/autograd.py

```diff
@@ -265,3 +265,60 @@ def float_precision(self):
         if isinstance(self.child, torch.Tensor) and not self.child.is_wrapper:
             return self.child
         return self
+
```

Comment:
Can you add tests for this?

It would be great to have simple tests for serialize / deserialize and also a test where you:

- train a model (with some toy data)
- run inference in the model (with another batch of toy data)
- serialize and deserialize the model
- check that the evaluation has the same result as the original model
- continue training the model and check that the weights are changing

Reply:
Thanks for the comments. I tried to perform training of an RNN using Autogradtensors just yesterday, and I bumped into some issues related to backpropagation yielding errors, because tensors were wrapped into AutogradTensors. I fixed this issue by extracting the tensors via an "hotfix" into the Pytorch's library code at `torch/autograd/__init__.py` in the `def backward` method.

At line 87, we just had:

```python grad_tensors = list(grad_tensors)```

I replaced that line  with:

```python
 try:
            grad_tensors = list(grad_tensors)
        except:
            print("Gradients extracted!")
            #extract the tensor from the autogradTensor in every tensor
            grad_tensors = extract_grads_from_autograd_tensors(grad_tensors)

def extract_grads_from_autograd_tensors(auto_grad_tensors):
    grad_tensors = []
    for i in range(len(auto_grad_tensors)):
        #if the child was not yet extracted, then do extract it
        
        tensor_grad = auto_grad_tensors[i].child
        grad_tensors.append(tensor_grad)
        
    return(grad_tensors)
```










Any suggestions on how this hotfix could be introduced into PySyft for autograd_training with Autogradtensors?

Reply:
What's the exception you get when you try to execute `list(grad_tensors)`?

Reply:
Probably adding these changes to hook should do the job, and in my opinion, is not that hacky.

Reply:
Yeah, hooking these changes should probably be sufficient. I just didn't have much of a clue as to where they should be placed.

Reply:
The error I was getting is the following one, BTW:

![image](https://user-images.githubusercontent.com/4907418/61272661-ee3f1780-a7a7-11e9-87f9-44ebed405875.png)